File: apps/android/feature/profile/data/src/main/java/com/scribblefit/feature/profile/data/repository/UserRepositoryImpl.kt
package com.scribblefit.feature.profile.data.repository

import com.scribblefit.feature.ledger.domain.repository.LedgerRepository
import com.scribblefit.feature.profile.domain.model.UserStats
import com.scribblefit.feature.profile.domain.repository.UserRepository
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class UserRepositoryImpl @Inject constructor(
    private val ledgerRepository: LedgerRepository
) : UserRepository {

    override fun getUserStats(): Flow<UserStats> =
        ledgerRepository.getWorkoutHistory().map { history ->
            UserStats(
                totalWorkouts = history.size,
                lifetimeVolume = history.sumOf { workout ->
                    workout.exercises.sumOf { exercise ->
                        exercise.sets.sumOf { set ->
                            set.weight * set.reps
                        }
                    }
                },
                prCount = 0,
                joinDate = history.minOfOrNull { it.date }
            )
        }
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/Profile/Data/Repository/UserRepositoryImpl.swift
import Foundation

public final class UserRepositoryImpl: UserRepository {
    private let ledgerRepository: LedgerRepository

    public init(ledgerRepository: LedgerRepository) {
        self.ledgerRepository = ledgerRepository
    }

    public func getUserStats() async throws -> UserStats {
        let history = try await ledgerRepository.getWorkoutHistory()

        let totalVolume = history.reduce(0.0) { $0 + $1.totalVolume }
        let firstWorkoutDate = history.map { $0.date }.min() ?? Date()

        return UserStats(
            totalWorkouts: history.count,
            lifetimeVolume: totalVolume,
            prCount: 0, // Placeholder
            joinDate: firstWorkoutDate
        )
    }
}
